### PR TITLE
Add admin menu for starting a game

### PR DESCRIPTION
**Pull request by RileyEv**
I have added an admin menu for starting a game. It is accessible from the main menu. It should probably be protected in the future to stop someone opening it on games day.

- Selecting admin on the menu connects you as a 'special' client with admin powers, which allow you to control the server. 
- Currently it consists of a button to start a game and a button to disconnect the client.
- Only one admin client is allowed to connect at a time to the server.
- To avoid needing 3 unity clients for development I have also added a catch that starts up the UI if running on `UNITY_EDITOR`, with the escape menu disabled and only allowing you to view the admin menu.

### CODE REVIEW

File: ProjectFiles/Assets/Scripts/Game/ServerGameLoop.cs [hunk 1]
```diff
@@ -53,10 +65,6 @@ public void Shutdown()
 
         public void Update()
         {
-            if(Input.GetKeyDown("a") && Input.GetKeyDown("b") && !roundManager.GameInProgress)
-            {
-                roundManager.StartGame();
-            }
```
Comment (ChrisGora) [inline ProjectFiles/Assets/Scripts/Game/ServerGameLoop.cs:59]: :heart:

File: ProjectFiles/Assets/Scripts/Network/EventType.cs [hunk 2]
```diff
@@ -5,17 +5,21 @@ namespace Network
...
+        ClientSpaceExitEvent        = 0x84,
+        
+        AdminClientStartGameEvent   = 0xA0,
+        
+        ServerHandshake             = 0x01,
+        ServerLocationUpdate        = 0x02,
+        ServerSpawnPlayerEvent      = 0x03,
+        ServerPreRoundStartEvent    = 0x04,
+        ServerRoundStartEvent       = 0x05,
+        ServerRoundEndEvent         = 0x06,
```
Comment (ChrisGora) [inline ProjectFiles/Assets/Scripts/Network/EventType.cs:22]: This aligned stuff will be gone next time I press 'reformat' ;p

File: ProjectFiles/Assets/Scripts/UI/UIController.cs [hunk 3]
```diff
@@ -56,9 +60,24 @@ private void Update()
...
+        }
 
+        public void SubscribeTriggerGameStartEvent(TriggerGameStartDelegate handler)
+        {
+            adminMenu.TriggerGameStartEvent += handler;
         }
 
+        public void OnGameStart(ushort freeRoamLength, ushort nPlayers)
+        {
+            Debug.Log("testing2");
```
Comment (ChrisGora) [inline ProjectFiles/Assets/Scripts/UI/UIController.cs]: Remove before you merge

File: ProjectFiles/Assets/Scripts/UI/UIController.cs [hunk 4]
```diff
@@ -56,9 +60,24 @@ private void Update()
...
+            if (ClientConfig.GameMode == GameMode.AdminMode || IsServerMode)
+            {
+                adminMenu.OnGameStart();
+                Debug.Log("testing");
```
Comment (ChrisGora) [inline ProjectFiles/Assets/Scripts/UI/UIController.cs]: Remove before you merge

**Review by ChrisGora — CHANGES_REQUESTED** [review]
You left some Debug.Logs in there. It looks really good otherwise

File: ProjectFiles/Assets/Scripts/Network/EventType.cs (hunk 2 above)
Comment (RileyEv) [reply ProjectFiles/Assets/Scripts/Network/EventType.cs:22]: And itll go back when I next use rider :smile: 

https://blog.jetbrains.com/dotnet/2017/11/28/aligning-code-columns-resharper-rider/

File: ProjectFiles/Assets/Scripts/UI/UIController.cs (hunk 3 above)
Comment (RileyEv) [reply ProjectFiles/Assets/Scripts/UI/UIController.cs]: whoops :man_facepalming:

File: ProjectFiles/Assets/Scripts/UI/UIController.cs (hunk 4 above)
Comment (RileyEv) [reply ProjectFiles/Assets/Scripts/UI/UIController.cs]: whoops :man_facepalming:

**Review by Dimmitsaras — APPROVED** [review]

**Review by Dimmitsaras — APPROVED** [review]

**Pull request merged**